src/core/Akka/Util/Internal/Extensions.cs

```diff
@@ -72,10 +72,30 @@ public static T Head<T>(this IEnumerable<T> self)
         /// <returns>The path elements</returns>
         public static IEnumerable<string> SplitDottedPathHonouringQuotes(this string path)
         {
-            return path.Split('\"')
-                .AlternateSelectMany(
-                    outsideQuote => outsideQuote.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries),
-                    insideQuote => new[] { insideQuote });
+            var i = 0;
+            var j = 0;
+            var result = new List<string>(8);
```

Comment:
I'm just liking that digit. ;) But can change to 4 according to your heuristics.